File: cs/Refactor2/SRP_DI_Workshop/Domain/OrderOperations/GetOrderOperation.cs
using System;
using SRP_DI_Workshop.ServiceInterface;

namespace SRP_DI_Workshop.Domain.OrderOperations
{
    public class GetOrderOperation : IOrderOperation
    {
        private readonly IOrderRepository _orderRepository;
        private readonly ILoggerService _loggerService;
        private readonly IResponseMessageFactory _responseMessageFactory;

        public Operation Operation { get; private set; }

        public GetOrderOperation(
            IOrderRepository orderRepository,
            ILoggerService loggerService,
            IResponseMessageFactory responseMessageFactory)
        {
            _orderRepository = orderRepository;
            _loggerService = loggerService;
            _responseMessageFactory = responseMessageFactory;

            Operation = Operation.GetOrderDetails;
        }

        public ResponseMessage ExecuteOperation(RequestMessage request)
        {
            try
            {
                Order orderToGet = _orderRepository.GetOrder(request.OrderId);

                return _responseMessageFactory.CreateOrderQueryResponseMessage(request, orderToGet);
            }
            catch (Exception ex)
            {
                _loggerService.WriteLine("Exception during operation GetOrderDetails: " + ex, "GetOrderDetailsError");
                return _responseMessageFactory.CreateErrorResponseMessage(ex);
            }
        }
    }
}

File: java/Refactor1/src/main/java/no/bouvet/solid/srpdip/domain/orderoperations/GetOrderOperation.java
package no.bouvet.solid.srpdip.domain.orderoperations;

import no.bouvet.solid.srpdip.OrderRepository;
import no.bouvet.solid.srpdip.domain.Order;
import no.bouvet.solid.srpdip.messageinterface.RequestMessage;
import no.bouvet.solid.srpdip.messageinterface.ResponseMessage;
import no.bouvet.solid.srpdip.messageinterface.ResponseMessageFactory;

public class GetOrderOperation implements OrderOperation {

	private OrderRepository orderRepository;

	private ResponseMessageFactory responseMessageFactory = new ResponseMessageFactory();

	public GetOrderOperation(OrderRepository orders) {
		this.orderRepository = orders;
	}

	@Override
	public ResponseMessage execute(RequestMessage request) {
		Order orderToGet = orderRepository.getOrder(request.getOrderId());

		return responseMessageFactory.createOrderQueryResponseMessage(request, orderToGet);
	}

}
